profiles/schema.py

```diff
@@ -1645,10 +1647,13 @@ def resolve_profile_with_access_token(self, info, **kwargs):
         return token.profile
 
     def resolve_service_connection_with_user_id(self, info, **kwargs):
-        return ServiceConnection.objects.get(
-            profile__user__uuid=kwargs["user_id"],
-            service__client_ids__client_id=kwargs["service_client_id"],
-        )
+        try:
+            return ServiceConnection.objects.get(
+                profile__user__uuid=kwargs["user_id"],
+                service__client_ids__client_id=kwargs["service_client_id"],
+            )
+        except ServiceConnection.DoesNotExist:
```

Comment:
Is there a scenario where we would need to know if the client id exists in the first place? Like can the user somehow end up in Keycloak via a service that is not registered to Profile BE? Or was it already checked in Keycloak?